src/main/java/ogc/rs/processes/collectionAppending/CollectionAppendingProcess.java

```diff
@@ -0,0 +1,412 @@
+            LOGGER.debug(arg);
+        }
+        return cmdLine;
+    }
+
+    private Future<Void> mergeTempTableToCollection(JsonObject requestInput) {
+
+        Promise<Void> promise = Promise.promise();
+        String jobId = requestInput.getString("jobId");
+        String collectionId = requestInput.getString("collectionsDetailsTableId");
+        String tempTableName = "temp_table_for_" + jobId;
+        LOGGER.debug("temp table name is : " + tempTableName);
+
+        getDbSchema(collectionId)
+                .onSuccess(columns -> {
+                    // Exclude 'id' column
+                    columns.remove("id");
+                    // Every column name should be put into double quotes to avoid the case sensitivity problem of Postgres
+                    String columnNames = columns.stream().map(s -> "\"" + s + "\"").collect(Collectors.joining(", "));
+                    String mergeQuery = String.format(
+                            MERGE_TEMP_TABLE_QUERY,
+                            collectionId,
+                            columnNames,
+                            columnNames,
+                            tempTableName
+                    );
+
+                    pgPool.withConnection(sqlConnection ->
+                            sqlConnection.query(mergeQuery)
+                                    .execute()
+                                    .onSuccess(successHandler -> {
+                                        LOGGER.debug("Merged temp table into main table successfully.");
+                                        promise.complete();
+                                    })
+                                    .onFailure(failureHandler -> {
+                                        LOGGER.error("Failed to merge temp table into main table: " + failureHandler.getMessage());
+                                        promise.fail(new ProcessException(500, "MERGE_FAILED", "Failed to merge temp table into main table."));
+                                    })
+                    );
+                })
+                .onFailure(promise::fail);
+
+        return promise.future();
+    }
+
+    private Future<Void> deleteTempTable(JsonObject requestInput) {
+
+        Promise<Void> promise = Promise.promise();
+        String jobId = requestInput.getString("jobId");
+        String tempTableName = "temp_table_for_" + jobId;
+        String deleteQuery = String.format(DELETE_TEMP_TABLE_QUERY, tempTableName);
+
+        pgPool.withConnection(sqlConnection ->
+                sqlConnection.query(deleteQuery)
+                        .execute()
+                        .onSuccess(successHandler -> {
+                            LOGGER.debug("Temporary table deleted successfully.");
+                            promise.complete();
+                        })
+                        .onFailure(failureHandler -> {
+                            LOGGER.error("Failed to delete temporary table: " + failureHandler.getMessage());
```

Comment:
Change this to a `FATAL` level log so that we know something critical has gone wrong.

Reply:
Changed the log level to fatal.

https://github.com/datakaveri/ogc-resource-server/pull/139/commits/0b83a45d78912dc598a3a5bbc2a271cf256b6f61